snmp/datadog_checks/snmp/snmp.py

```diff
@@ -107,15 +107,18 @@ def _get_profiles_mapping(self):
         """
         profiles_by_oid = {}  # type: Dict[str, str]
         for name, profile in self.profiles.items():
-            sys_object_oid = profile['definition'].get('sysobjectid')
-            if sys_object_oid is not None:
-                profile_match = profiles_by_oid.get(sys_object_oid)
-                if profile_match:
-                    raise ConfigurationError(
-                        "Profile {} has the same sysObjectID ({}) as {}".format(name, sys_object_oid, profile_match)
-                    )
-                else:
-                    profiles_by_oid[sys_object_oid] = name
+            sys_object_oids = profile['definition'].get('sysobjectid')
+            if sys_object_oids is not None:
```

Comment:
(Nit) Consider a guard clause to reduce the indentation here (it's becoming a bit hard to read IMO):

```suggestion
            if sys_object_oids is None:
                continue
            
            ...
```